src/components/tooltip/tooltip.jsx

```diff
@@ -117,28 +118,58 @@ class Tooltip extends React.Component {
   }
 
   renderPopup(content, placement, tooltipStyle) {
-    const style = {
+    if (this.container && this.left === null && (placement === 'below' || placement === 'above')) {
+      const popupRect = this.popup.getBoundingClientRect();
+      if (popupRect.left < 0) {
+        this.left = 10 - popupRect.left;
+      } else if (popupRect.right > window.innerWidth) {
+        this.left = -(popupRect.right - window.innerWidth + 10);
+      }
+    }
+
+    let arrowAfter;
+    let arrowBefore;
+    if (placement === 'below') {
+      arrowBefore = <div className={this.classes.belowArrow} />;
+    } else if (placement === 'above') {
+      arrowAfter = <div className={this.classes.aboveArrow} />;
+    } else if (placement === 'left') {
+      arrowAfter = <div className={this.classes.leftArrow} />;
+    } else if (placement === 'right') {
+      arrowAfter = <div className={this.classes.rightArrow} />;
+    }
+
+    const wrapperStyle = {
       opacity: this.state.hover || this.state.toggled ? 1 : 0,
       pointerEvents: this.state.hover || this.state.toggled ? 'all' : 'none',
       ...tooltipStyle,
     };
 
     if (this.props.fixedWidth) {
-      style.width = this.props.fixedWidth;
-      style.whiteSpace = 'inherit';
+      wrapperStyle.width = this.props.fixedWidth;
+      wrapperStyle.whiteSpace = 'inherit';
```

Comment:
These two can be set in the styles file instead since they only rely on props:

```
popup: {
  width: ({ fixedWidth }) => fixedWidth || 'initial',
  whiteSpace: ({ fixedWidth }) => fixedWidth ? 'inherit' : 'no-wrap',
 // ...
}
```